plugins/modules/gcp_secret_manager.py

```diff
@@ -0,0 +1,429 @@
+    type: str
+  payload:
+    description:
+    - The base 64 secret payload including CRC for validation
+    retunred: success
+    type: dict
+'''
+
+################################################################################
+# Imports
+################################################################################
+
+from ansible_collections.google.cloud.plugins.module_utils.gcp_utils import (
+    navigate_hash,
+    GcpSession,
+    GcpModule,
+    GcpRequest,
+    remove_nones_from_dict,
+    replace_resource_dict,
+)
+
+import json
+# for decoding and validating secrets
+import base64
+import binascii
+
+
+def get_auth(module):
+    return GcpSession(module, 'secret-manager')
+
+def self_access_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}/versions/{calc_version}:access".format(**module.params)
+
+def self_get_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}/versions/{calc_version}".format(**module.params)
+
+def self_update_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}/versions/{calc_version:version}".format(**module.params)
+
+def self_list_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}/versions?filter=state:ENABLED".format(**module.params)
+
+def self_delete_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}".format(**module.params)
+
+
+def fetch_resource(module, allow_not_found=True):
+    auth = get_auth(module)
+    # set version to the latest version because
+    # we can't be sure that "latest" is always going 
+    # to be set if secret versions get disabled
+    # see https://issuetracker.google.com/issues/286489671
+    if module.params['version'] == "latest" or module.params['version'] == 'all':
+        version_list = list_secret_versions(module)
+        latest_version = None
+        if version_list is None:
+            return None
+        
+        if "versions" in version_list:
+            latest_version = sorted(version_list['versions'], key=lambda d: d['name'])[-1]['name'].split('/')[-1]
+            module.params['calc_version'] = latest_version
+        else:
+            # if this occurs, there are no available secret versions
+            # handle the corner case that we tried to delete
+            # a secret version that doesn't exist
+            if module.params['state'] == "absent":
+                return { "action": "delete_secret" }
+
+    link = self_access_link(module)
+    access_obj = return_if_object(module, auth.get(link), allow_not_found)
+    if access_obj is None:
+        return None
+    link = self_get_link(module)
+    get_obj = return_if_object(module, auth.get(link), allow_not_found)
+    if get_obj is None:
+        return None
+    return merge_dicts(get_obj, access_obj)
+
+def merge_dicts(x, y):
+    z = x.copy()
```

Comment:
note that this is only a shallow copy (nested objects will still be modified) - is that ok? on first glance it seems like it's fine, but just checking.

Reply:
Nice catch,  I don't see why it wouldn't be okay to do the shallow copy here but now that you've brought it up, I feel like I should do a deep copy :D.